tests/runner/test_utils.py

```diff
@@ -0,0 +1,119 @@
+from unittest.mock import Mock
+
+import pytest
+
+from prefect.runner.utils import (
+    inject_schemas_into_openapi,
+    merge_definitions,
+    update_refs_in_schema,
+    update_refs_to_components,
+)
+
+
+@pytest.fixture
+def mock_app():
+    app = Mock()
+    app.routes = [Mock(name="dummy_route", methods=["GET"], path="/dummy")]
+    return app
+
+
+@pytest.fixture
+def deployment_schemas():
+    return {
+        "deployment1": {
+            "definitions": {
+                "Model1": {
+                    "type": "object",
+                    "properties": {"field1": {"type": "string"}},
+                }
+            }
+        },
+        "deployment2": {
+            "definitions": {
+                "Model2": {
+                    "type": "object",
+                    "properties": {"field2": {"type": "integer"}},
+                }
+            }
+        },
+    }
+
+
+@pytest.fixture
+def schema_item():
+    return {"$ref": "#/definitions/Model1"}
+
+
+@pytest.fixture
+def deployment_openapi_schema(schema_item):
+    return {
+        "paths": {
+            "/path": {
+                "get": {
+                    "requestBody": {
+                        "content": {"application/json": {"schema": schema_item}}
+                    }
+                }
+            }
+        }
+    }
+
+
+@pytest.fixture
+def openapi_schema():
+    return {"components": {"schemas": {}}}
+
+
+def test_inject_schemas_into_openapi(mock_app, deployment_schemas):
+    augmented_schema = inject_schemas_into_openapi(mock_app, deployment_schemas)
+    assert "Model1" in augmented_schema["components"]["schemas"]
+    assert "Model2" in augmented_schema["components"]["schemas"]
+    assert augmented_schema["components"]["schemas"]["Model1"]["type"] == "object"
+    assert augmented_schema["components"]["schemas"]["Model2"]["type"] == "object"
+    assert (
+        augmented_schema["components"]["schemas"]["Model1"]["properties"]["field1"][
+            "type"
+        ]
+        == "string"
+    )
+    assert (
+        augmented_schema["components"]["schemas"]["Model2"]["properties"]["field2"][
+            "type"
+        ]
+        == "integer"
+    )
+
+
+def test_merge_definitions(deployment_schemas, openapi_schema):
+    merge_definitions(deployment_schemas, openapi_schema)
+    assert "Model1" in openapi_schema["components"]["schemas"]
+    assert "Model2" in openapi_schema["components"]["schemas"]
+    assert openapi_schema["components"]["schemas"]["Model1"]["type"] == "object"
+    assert openapi_schema["components"]["schemas"]["Model2"]["type"] == "object"
+    assert (
+        openapi_schema["components"]["schemas"]["Model1"]["properties"]["field1"][
+            "type"
+        ]
+        == "string"
+    )
+    assert (
+        openapi_schema["components"]["schemas"]["Model2"]["properties"]["field2"][
+            "type"
+        ]
+        == "integer"
+    )
```

Comment:
If I were looking at only these tests, I'd assume these two functions do exactly the same thing. Are the outputs deterministic enough to compare the output directly to an expected dictionary?

Reply:
cleaned up the tests a bit in [17ff57f](https://github.com/PrefectHQ/prefect/pull/11333/commits/17ff57ff5733f63ab159e5354002e69f7b7eee4f), let me know what you think!